### ISSUE TITLE
backtest should check if Hyperopt file (.json) changed

### ISSUE THREAD

**Issue by nateemma** [freqtrade/freqtrade#6735]
## Describe your environment

  * Operating system: MacOS
  * Python Version: 3.9.12 (`python -V`)
  * CCXT version: 1.80.61 (`pip freeze | grep ccxt`)
  * Freqtrade Version: freqtrade develop-44000ae0b (`freqtrade -V` or `docker-compose run --rm freqtrade -V` for Freqtrade running in docker)
  
Note: All issues other than enhancement requests will be closed without further comment if the above template is deleted or not filled out.

## Describe the problem:

the _backtesting_ command just added caching of test results, which is nice, but it doesn't check whether the Hyperopt file (_strategy_.json)  has changed. This means that if I run _hyperopt_ followed by _backtesting_ (or manually tweak the parameters), it will use the 'old' backtest results.

I know I can disable this by specifying _--cache none_, but it would be really nice to take advantage of caching where it is valid.

### Steps to reproduce:

  1. run _freqtrade backtesting_
  2. run _freqtrade hyperopt_ on the same strategy, or change _strategy_.json manually
  3. run _freqtrade backtesting_ again
  
### Observed Results:

  * The final _freqtrade backtesting_ command will use the cached backtesting results from the first run
  * It should recognize that the hyper parameters have changed and re-run the tests. Note that it *does* do this if the strategy itself is changed, so there is some checking going on.

### Relevant code exceptions or logs

- Normal run:

  ```
freqtrade backtesting   --timerange=20211029-20220427 -c user_data/strategies/kucoin/config_kucoin.json --strategy-path user_data/strategies/kucoin --strategy-list FFT2

2022-04-27 11:09:04,014 - freqtrade.configuration.load_config - INFO - Using config: user_data/strategies/kucoin/config_kucoin.json ...
2022-04-27 11:09:04,015 - freqtrade.loggers - INFO - Verbosity set to 0
2022-04-27 11:09:04,015 - freqtrade.configuration.configuration - INFO - Using additional Strategy lookup path: user_data/strategies/kucoin
2022-04-27 11:09:04,015 - freqtrade.configuration.configuration - INFO - Using max_open_trades: 3 ...
2022-04-27 11:09:04,015 - freqtrade.configuration.configuration - INFO - Parameter --timerange detected: 20211029-20220427 ...
2022-04-27 11:09:04,015 - freqtrade.configuration.configuration - INFO - Using user-data directory: /Users/philprice/freqtrade/user_data ...
2022-04-27 11:09:04,015 - freqtrade.configuration.configuration - INFO - Using data directory: /Users/philprice/freqtrade/user_data/data/kucoin ...
2022-04-27 11:09:04,015 - freqtrade.configuration.configuration - INFO - Using strategy list of 1 strategies
2022-04-27 11:09:04,015 - freqtrade.configuration.configuration - INFO - Parameter --cache=day detected ...
2022-04-27 11:09:04,015 - freqtrade.configuration.check_exchange - INFO - Checking exchange...
2022-04-27 11:09:04,017 - freqtrade.configuration.check_exchange - WARNING - Exchange "kucoin" is known to the the ccxt library, available for the bot, but not officially supported by the Freqtrade development team. It may work flawlessly (please report back) or have serious issues. Use it at your own discretion.
2022-04-27 11:09:04,017 - freqtrade.configuration.configuration - INFO - Using pairlist from configuration.
2022-04-27 11:09:04,018 - freqtrade.configuration.config_validation - INFO - Validating configuration ...
2022-04-27 11:09:04,019 - freqtrade.commands.optimize_commands - INFO - Starting freqtrade in Backtesting mode
2022-04-27 11:09:04,019 - freqtrade.exchange.exchange - INFO - Instance is running with dry_run enabled
2022-04-27 11:09:04,019 - freqtrade.exchange.exchange - INFO - Using CCXT 1.80.61
2022-04-27 11:09:04,019 - freqtrade.exchange.exchange - INFO - Applying additional ccxt config: {'enableRateLimit': False, 'urls': {'api': {'public': 'http://127.0.0.1:8200/kucoin', 'private': 'http://127.0.0.1:8200/kucoin'}}}
2022-04-27 11:09:04,021 - freqtrade.exchange.exchange - INFO - Applying additional ccxt config: {'enableRateLimit': False, 'urls': {'api': {'public': 'http://127.0.0.1:8200/kucoin', 'private': 'http://127.0.0.1:8200/kucoin'}}}
2022-04-27 11:09:04,023 - freqtrade.exchange.exchange - INFO - Using Exchange "KuCoin"
2022-04-27 11:09:04,159 - freqtrade.resolvers.exchange_resolver - INFO - Using resolved exchange 'Kucoin'...
2022-04-27 11:09:04,210 - freqtrade.resolvers.iresolver - INFO - Using resolved strategy FFT2 from '/Users/philprice/freqtrade/user_data/strategies/kucoin/FFT2.py'...
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Loading parameters from file /Users/philprice/freqtrade/user_data/strategies/kucoin/FFT2.json
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: buy_fft_cutoff = 0.31
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: buy_fft_dev = 2.0
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: buy_fft_diff = 2.0
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: csell_endtrend_respect_roi = False
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: csell_pullback = True
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: csell_pullback_amount = 0.02
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: csell_pullback_respect_roi = False
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: csell_roi_end = 0.006
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: csell_roi_start = 0.045
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: csell_roi_time = 753
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: csell_roi_type = step
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: csell_trend_type = none
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: cstop_bail_how = none
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: cstop_bail_roc = -1.196
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: cstop_bail_time = 363
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: cstop_bail_time_trend = True
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: cstop_loss_threshold = -0.022
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: cstop_max_stoploss = -0.138
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: sell_fft_dev = -2.0
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - Strategy Parameter: sell_fft_diff = -2.0
2022-04-27 11:09:04,210 - freqtrade.strategy.hyper - INFO - No params for protection found, using default values.

2022-04-27 11:09:04,210 - freqtrade.resolvers.strategy_resolver - WARNING - DEPRECATED: Using 'sell_profit_only' moved to 'exit_profit_only'.
2022-04-27 11:09:04,210 - freqtrade.resolvers.strategy_resolver - WARNING - DEPRECATED: Using 'use_sell_signal' moved to 'use_exit_signal'.
2022-04-27 11:09:04,210 - freqtrade.resolvers.strategy_resolver - WARNING - DEPRECATED: Using 'ignore_roi_if_buy_signal' moved to 'ignore_roi_if_entry_signal'.
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'timeframe' with value in config file: 5m.
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_currency' with value in config file: USDT.
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_amount' with value in config file: 3300.
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'unfilledtimeout' with value in config file: {'entry': 10, 'exit': 10, 'exit_timeout_count': 0, 'unit': 'minutes'}.
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using minimal_roi: {'0': 0.1}
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using timeframe: 5m
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stoploss: -0.1
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop: False
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop_positive_offset: 0.0
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_only_offset_is_reached: False
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using use_custom_stoploss: True
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using process_only_new_candles: True
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_types: {'entry': 'limit', 'exit': 'limit', 'stoploss': 'limit', 'stoploss_on_exchange': False, 'stoploss_on_exchange_interval': 60}
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_time_in_force: {'entry': 'gtc', 'exit': 'gtc'}
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_currency: USDT
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_amount: 3300
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using protections: []
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using startup_candle_count: 128
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using unfilledtimeout: {'entry': 10, 'exit': 10, 'exit_timeout_count': 0, 'unit': 'minutes'}
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using use_exit_signal: True
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using exit_profit_only: False
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ignore_roi_if_entry_signal: True
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using exit_profit_offset: 0.0
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using disable_dataframe_checks: False
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ignore_buying_expired_candle_after: 0
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using position_adjustment_enable: False
2022-04-27 11:09:04,211 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using max_entry_position_adjustment: -1
2022-04-27 11:09:04,211 - freqtrade.configuration.config_validation - INFO - Validating configuration ...
2022-04-27 11:09:04,213 - freqtrade.resolvers.iresolver - INFO - Using resolved pairlist StaticPairList from '/Users/philprice/freqtrade/freqtrade/plugins/pairlist/StaticPairList.py'...
2022-04-27 11:09:04,214 - freqtrade.resolvers.iresolver - INFO - Using resolved pairlist SpreadFilter from '/Users/philprice/freqtrade/freqtrade/plugins/pairlist/SpreadFilter.py'...
2022-04-27 11:09:04,216 - freqtrade.resolvers.iresolver - INFO - Using resolved pairlist PriceFilter from '/Users/philprice/freqtrade/freqtrade/plugins/pairlist/PriceFilter.py'...
2022-04-27 11:09:04,217 - freqtrade.resolvers.iresolver - INFO - Using resolved pairlist RangeStabilityFilter from '/Users/philprice/freqtrade/freqtrade/plugins/pairlist/rangestabilityfilter.py'...
2022-04-27 11:09:04,217 - freqtrade.resolvers.iresolver - INFO - Using resolved pairlist VolatilityFilter from '/Users/philprice/freqtrade/freqtrade/plugins/pairlist/VolatilityFilter.py'...
2022-04-27 11:09:04,608 - freqtrade.data.history.history_utils - INFO - Using indicator startup period: 128 ...
:
:
2022-04-27 11:09:11,086 - freqtrade.optimize.backtesting - INFO - Backtesting with data from 2021-10-29 00:00:00 up to 2022-04-25 23:15:00 (178 days).
2022-04-27 11:09:54,707 - freqtrade.misc - INFO - dumping json to "/Users/philprice/freqtrade/user_data/backtest_results/backtest-result-2022-04-27_11-09-54.meta.json"
2022-04-27 11:09:54,708 - freqtrade.misc - INFO - dumping json to "/Users/philprice/freqtrade/user_data/backtest_results/backtest-result-2022-04-27_11-09-54.json"
2022-04-27 11:09:54,725 - freqtrade.misc - INFO - dumping json to "/Users/philprice/freqtrade/user_data/backtest_results/.last_result.json"

  ```

- run again, without changing anything, uses the cached results (correct):

```
freqtrade backtesting   --timerange=20211029-20220427 -c user_data/strategies/kucoin/config_kucoin.json --strategy-path user_data/strategies/kucoin --strategy-list FFT2
:
:
2022-04-27 11:18:08,089 - freqtrade.data.btanalysis - INFO - Loading backtest result from /Users/philprice/freqtrade/user_data/backtest_results/backtest-result-2022-04-27_11-09-54.json
2022-04-27 11:18:08,092 - freqtrade.optimize.backtesting - INFO - Reusing result of previous backtest for FFT2
```

- change hyperparam  FFT2.json (I changed buy_fft_dev from 2.0 to 3.0), it uses the cached results again 
(incorrect):

```
freqtrade backtesting   --timerange=20211029-20220427 -c user_data/strategies/kucoin/config_kucoin.json --strategy-path user_data/strategies/kucoin --strategy-list FFT2
:
:
2022-04-27 11:19:44,400 - freqtrade.strategy.hyper - INFO - Loading parameters from file /Users/philprice/freqtrade/user_data/strategies/kucoin/FFT2.json
2022-04-27 11:19:44,400 - freqtrade.strategy.hyper - INFO - Strategy Parameter: buy_fft_cutoff = 0.31
2022-04-27 11:19:44,400 - freqtrade.strategy.hyper - INFO - Strategy Parameter: buy_fft_dev = 3.0
2022-04-27 11:19:44,400 - freqtrade.strategy.hyper - INFO - Strategy Parameter: buy_fft_diff = 2.0
:
:
2022-04-27 11:19:46,791 - freqtrade.data.btanalysis - INFO - Loading backtest result from /Users/philprice/freqtrade/user_data/backtest_results/backtest-result-2022-04-27_11-09-54.json
2022-04-27 11:19:46,794 - freqtrade.optimize.backtesting - INFO - Reusing result of previous backtest for FFT2

```

- change strategy file (I just added some lines that do nothing), it does *not* use the cached results (correct):

```
freqtrade backtesting   --timerange=20211029-20220427 -c user_data/strategies/kucoin/config_kucoin.json --strategy-path user_data/strategies/kucoin --strategy-list FFT2
:
:
2022-04-27 11:22:24,740 - freqtrade.strategy.hyper - INFO - Strategy Parameter: buy_fft_dev = 3.0
:
:
2022-04-27 11:22:27,992 - freqtrade.optimize.backtesting - INFO - Loading data from 2021-10-28 13:20:00 up to 2022-04-25 23:15:00 (179 days).
2022-04-27 11:22:27,992 - freqtrade.optimize.backtesting - INFO - Dataload complete. Calculating indicators
2022-04-27 11:22:27,996 - freqtrade.optimize.backtesting - INFO - Running backtesting for Strategy FFT2
```

**Comment by xmatthias** [comment]
Great catch :+1: 

Should be pretty easy to fix.
